lib/utils/validateFormData.ts

```diff
@@ -0,0 +1,46 @@
+import { IUser } from "lib/db/models/users";
```

Comment:
Does NextAuth provide anything out-of-the-box for this?

Reply:
We also have a validator package in the deps that we could reuse

Reply:
Well next-auth can validate the payload on server side. But for better user experience, we should show the basic errors from client side. This way there will be less fetch calls on the server and user receives immediate feedback.